ext/page_log/palite/palite.cpp

```diff
@@ -26,6 +26,100 @@
  * OTHER DEALINGS IN THE SOFTWARE.
  */
 
+/*-
+ * -= How PALite works =-
+ *
+ * PALite implements PALI (the "Page Log Interface") using SQLite as the
+ * underlying storage engine.
+ *
+ * PALite consists of the three main storage table types:
+ *  - Globals: stores global metadata (last LSN, known table IDs)
+ *  - Checkpoints: stores checkpoint information
+ *  - Pages: stores page data (full pages and deltas)
+ *
+ * Each storage table type (Globals, Checkpoints, Pages) is implemented as a
+ * separate SQLite database file:
+ *  - globals.db
+ *  - checkpoints.db
+ *  - pages_[N].db  - one per WT table (N is the table ID)
+ *
+ * -= Concurrency model =-
+ *
+ * SQLite supports multiple simultaneous read transactions coming from separate
+ * database connections, possibly in separate threads or processes, but only one
+ * simultaneous write transaction.
+ *
+ * As a result, PALite uses a readers-writer lock at the table level to ensure
+ * that read operations can occur concurrently while write operations are
+ * serialized.
```

Comment:
Should we also briefly mention the storage-wide reader-writer lock that we have?